src/openms_gui/source/VISUAL/Spectrum1DCanvas.cpp

```diff
@@ -1199,10 +1199,6 @@ namespace OpenMS
     }
 
     // sort spectra in ascending order of position (ensure that we sort all spectra as well as the currently
-    for (Size i = 0; i < getCurrentLayer_().getPeakData()->size(); ++i)
-    {
-      (*getCurrentLayer_().getPeakDataMuteable())[i].sortByPosition();
-    }
     getCurrentLayer_().sortCurrentSpectrumByPosition();
```

Comment:
I get why we need to sort the current spectrum, but why are we sorting all spectra?

Reply:
I think it is a good idea to have the spectra sorted by m/z. Doing this during loading seems as the best option to me. I guess that code did not have much of a performance impact for older instruments.

Reply:
I think the issue is that I have ion mobility data and that means sorting involves sorting the float data array as well. First, each spectrum has many more data points and second the code for sorting is much more involved. We can easily have 30k or more data points per spectrum

Reply:
I have now replaced this with an `is_sorted` check which also solves the issue for me